certbot/main.py

```diff
@@ -366,6 +367,50 @@ def _init_le_client(config, authenticator, installer):
     return client.Client(config, acc, authenticator, installer, acme=acme)
 
 
+def register(config, unused_plugins):
+    """Create or modify accounts on the server."""
+
+    # Portion of _determine_account logic to see whether accounts already
+    # exist or not.
+    account_storage = account.AccountFileStorage(config)
+    accounts = account_storage.find_all()
+
+    # registering a new account
+    if not config.update_registration:
+        if len(accounts) > 0:
+            # TODO: add a flag to register a duplicate account (this will
+            #       also require extending _determine_account's behavior
+            #       or else extracting the registration code from there)
+            return ("There is an existing account; registration of a "
+                    "duplicate account with this command is currently "
+                    "unsupported.")
+        # _determine_account will register an account
+        _determine_account(config)
+        return
+
+    # --update-registration
+    if len(accounts) == 0:
+        return "Could not find an existing account to update."
+    if config.email is None:
+        return ("Currently, --update-registration can only change the e-mail "
+                "address\nassociated with an account. A new e-mail address is "
+                "required\n(hint: --email)")
+    acc, acme = _determine_account(config)
+    acme_client = client.Client(config, acc, None, None, acme=acme)
+    try:
+        updated_reg = client.messages.Registration.from_data(email=config.email)
+        acme_client.acme.update_registration(acme_client.account.regr,
+                                             updated_reg)
+    except acme_errors.UnexpectedUpdate:
+        # We expect the unexpected update!
+        pass
+    query_data = acme_client.acme.query_registration(acme_client.account.regr)
+    # We rely on an ACME exception to interrupt this process if it didn't work.
+    print("Registration change succeeded. New registration data:\n")
+    print(query_data)
```

Comment:
Do we really want to print the entire registration resource here? It's kind of an ugly mess.

Since users can only use this to update their e-mail address, perhaps we can just say that their email address updated to `config.email`.

Also, in most of the client we don't call `print` directly in favor of using `IDisplay`/`IReporter`. Perhaps we should just register a message with `IReporter` about the update?